buildSrc/src/main/kotlin/org/gradle/plugins/buildscan/BuildScanConfigurationPlugin.kt

```diff
@@ -0,0 +1,209 @@
+                }
+            }
+        }
+    }
+
+    private
+    fun Project.extractCiOrLocalData() {
+        if (isCiServer) {
+            buildScan.tag("CI")
+            buildScan.tag(System.getenv("TEAMCITY_BUILDCONF_NAME"))
+            buildScan.link("TeamCity Build", System.getenv("BUILD_URL"))
+            buildScan.value("Build ID", System.getenv("BUILD_ID"))
+            setCommitId(System.getenv("BUILD_VCS_NUMBER"))
+        } else {
+            buildScan.tag("LOCAL")
+        }
+    }
+
+    private
+    fun Project.extractVcsData() {
+
+        fun Project.run(vararg args: String): String {
+            val process = ProcessBuilder(args.toList())
+                .directory(rootDir)
+                .start()
+            assert(process.waitFor() == 0)
+            return process.inputStream.bufferedReader().readText().trim()
+        }
+
+        fun Project.execAsync(f: Runnable) {
+            val latch = java.util.concurrent.CountDownLatch(1)
+            thread(start = true) {
+                try {
+                    f.run()
+                } catch (e: Exception) {
+                    rootProject.logger.warn("Build scan user data async exec failed", e)
+                } finally {
+                    latch.countDown()
+                }
+            }
+        }
+
+        execAsync(Runnable {
+            val commitId = run("git", "rev-parse", "--verify", "HEAD")
+            setCommitId(commitId)
+            val status = run("git", "status", "--porcelain")
+            if (!status.isEmpty()) {
+                buildScan {
+                    tag("dirty")
+                    value("Git Status", status)
+                }
+            }
+        })
+
+        execAsync(Runnable {
+            val branchName = run("git", "rev-parse", "--abbrev-ref", "HEAD")
+            if (!branchName.isEmpty() && branchName != "HEAD") {
+                buildScan.tag(branchName)
+                buildScan.value("Git Branch Name", branchName)
+            }
+        })
+    }
+
+    private fun Project.extractBuildCacheData() {
+        if (gradle.startParameter.isBuildCacheEnabled) {
+            buildScan.tag("CACHED")
+
+            val tasksToInvestigateForCacheWithPaths = if (project.hasProperty("cache.investigate.tasks.paths"))
+                project.property("cache.investigate.tasks.paths").toString().split(",") else listOf(":baseServices:classpathManifest")
+
+            val taskPropertiesWithFullFileSnapshot = mapOf(
+                ":baseServices:compileJava" to listOf("classpath"))
+
+            project.buildScan.buildFinished({
+                allprojects.map { it.tasks }.flatten().forEach {
+                    if (it.state.executed && (tasksToInvestigateForCacheWithPaths.contains(it.path)
+                            || taskPropertiesWithFullFileSnapshot.keys.contains(it.path))) {
+                        if (tasksToInvestigateForCacheWithPaths.contains(it.path)) {
```

Comment:
With the above changes, these two if statements can be simplified to:
```
if (it.state.executed && tasksWithClassloaderInformation.contains(it.path)) {
...
}
```

Reply:
If you could commit this change in conjunction with the other one that would be great.